File: 10/code-shapes.java
public abstract class Shape {
	private Point center;
	protected Shape(Point center) { /*...*/ }

	public abstract void draw();
	public abstract void fill();
	public void drawAndFill() {
		this.draw();
		this.fill();
	}
}

public class Circle extends Shape {
	private double radius;
	public Circle(Point center, double radius) { /*...*/ }

	public void draw() { /*...*/ }
	public void fill() { /*...*/ }
}

File: 10/code-shapes.js
function Shape(center) {
	this.center = center;
};
Shape.prototype.drawAndFill = function() {
	this.draw();
	this.fill();
};

function Circle(center, radius) {
	Shape.call(this, center);
	this.radius = radius;
}
/* $Клонирование прототипа.$ */
Circle.prototype = Object.create(Shape.prototype); 
Circle.prototype.draw = function() { /* ... */ };
Circle.prototype.fill = function() { /* ... */ };


